### PR TITLE
Make export config file downloads require login

### PR DESCRIPTION
**Pull request by czue**
https://dimagi-dev.atlassian.net/browse/SC-1154

### CODE REVIEW

**Review by proteusvacuum — APPROVED** [review]

**Comment by czue** [comment]
@proteusvacuum added a few commits to support the versioning added by @Rohit25negi - can you review again?

File: apps/exports/views.py [hunk 1]
```diff
@@ -135,18 +136,24 @@ def export_details(request, export_id):
...
-    response = HttpResponse(export.config_file.read(), content_type='application/force-download')
-    response['Content-Disposition'] = f'attachment; filename={os.path.basename(export.config_file.name)}'
+@login_required
+def download_export_file_version(request, version_id):
+    version = get_object_or_404(Version, id=version_id)
+    return _download_config_file(version.field_dict['config_file'])
+
+
+def _download_config_file(export_file_field):
+    response = HttpResponse(export_file_field.read(), content_type='application/force-download')
```
Comment (proteusvacuum) [inline apps/exports/views.py:155]: Out of curiosity, why this mimetype? I think elsewhere (in HQ) we use just the `Content-Disposition` header to show the user the download box which seems to work fine.

**Review by proteusvacuum — APPROVED** [review]

File: apps/exports/views.py (hunk 1 above)
Comment (czue) [reply apps/exports/views.py:155]: Good question. My guess is I just copy/pasted it from here: https://stackoverflow.com/a/1158750/8207.

I guess the right thing to do would be to guess the content type from the file field, but do you think it's worth it?

Or we can just use [octet-stream](https://stackoverflow.com/questions/20508788/do-i-need-content-type-application-octet-stream-for-file-download) I suppose.

Comment (czue) [reply apps/exports/views.py:155]: will merge, but happy to update in a follow up.

Comment (proteusvacuum) [reply apps/exports/views.py:155]: I don't think a change is necessary - was just curious. Thanks for the explanation!

**Pull request merged**